velox/exec/tests/PrefixSortTest.cpp

```diff
@@ -300,30 +331,63 @@ TEST_F(PrefixSortTest, checkMaxNormalizedKeySizeForMultipleKeys) {
   // The normalizedKeySize for BIGINT should be 8 + 1.
   std::vector<TypePtr> keyTypes = {BIGINT(), BIGINT()};
   std::vector<CompareFlags> compareFlags = {kAsc, kDesc};
-  auto sortLayout = PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 8);
+  auto sortLayout =
+      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 8, 9);
   ASSERT_FALSE(sortLayout.hasNormalizedKeys);
 
   auto sortLayoutOneKey =
-      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 9);
+      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 9, 9);
   ASSERT_TRUE(sortLayoutOneKey.hasNormalizedKeys);
   ASSERT_TRUE(sortLayoutOneKey.hasNonNormalizedKey);
   ASSERT_EQ(sortLayoutOneKey.prefixOffsets.size(), 1);
   ASSERT_EQ(sortLayoutOneKey.prefixOffsets[0], 0);
 
   auto sortLayoutOneKey1 =
-      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 17);
+      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 17, 12);
   ASSERT_TRUE(sortLayoutOneKey1.hasNormalizedKeys);
   ASSERT_TRUE(sortLayoutOneKey1.hasNonNormalizedKey);
   ASSERT_EQ(sortLayoutOneKey1.prefixOffsets.size(), 1);
   ASSERT_EQ(sortLayoutOneKey1.prefixOffsets[0], 0);
 
   auto sortLayoutTwoKeys =
-      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 18);
+      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 18, 12);
   ASSERT_TRUE(sortLayoutTwoKeys.hasNormalizedKeys);
   ASSERT_FALSE(sortLayoutTwoKeys.hasNonNormalizedKey);
+  ASSERT_FALSE(
+      sortLayoutTwoKeys.comparisonStartIndex <
+      sortLayoutTwoKeys.numNormalizedKeys);
   ASSERT_EQ(sortLayoutTwoKeys.prefixOffsets.size(), 2);
   ASSERT_EQ(sortLayoutTwoKeys.prefixOffsets[0], 0);
   ASSERT_EQ(sortLayoutTwoKeys.prefixOffsets[1], 9);
 }
+
+TEST_F(PrefixSortTest, makeSortLayoutForString) {
+  std::vector<TypePtr> keyTypes = {VARCHAR(), BIGINT()};
+  std::vector<CompareFlags> compareFlags = {kAsc, kDesc};
+
+  auto sortLayoutOneKey =
+      PrefixSortLayout::makeSortLayout(keyTypes, compareFlags, 24, 12);
+  ASSERT_TRUE(sortLayoutOneKey.hasNormalizedKeys);
+  ASSERT_TRUE(sortLayoutOneKey.hasNonNormalizedKey);
+  ASSERT_TRUE(
+      sortLayoutOneKey.comparisonStartIndex <
+      sortLayoutOneKey.numNormalizedKeys);
+  ASSERT_EQ(sortLayoutOneKey.encodeSizes.size(), 1);
+  ASSERT_EQ(sortLayoutOneKey.encodeSizes[0], 13);
+
+  // Last key type is varbinary, we assume only partail data is encoded in
```

Comment:
typo partail -> partial, you could install extension Code Spell Checker in vscode, https://medium.com/@anticultist/setting-up-a-spell-checker-for-vs-code-8086462fce66